From 4c8b6a7b1f9fd3ddb9d7c8a8b70c2528dbc46e52 Mon Sep 17 00:00:00 2001
From: Nam Ninh <nam.ninh@windriver.com>
Date: Mon, 24 Nov 2014 10:13:28 -0500
Subject: [PATCH] kvm: handle read request of perf capabilities from MSR

On Canoe Pass, we occationally see this warning when we run a guest
on OVP host:
[91234.021762] kvm [3889]: vcpu0 unhandled rdmsr: 0x345

This request is to read the perf capabilities from MSR offset 0x345,
but the MSR read handler does not handle it, so we see the warning.
This patch simply reads the KVM native capabilities and returns it
to the guest.

Signed-off-by: Nam Ninh <nam.ninh@windriver.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index b7ce7f3f4362..136df7d76841 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -2454,6 +2454,9 @@ int kvm_get_msr_common(struct kvm_vcpu *vcpu, struct msr_data *msr_info)
 			return 1;
 		msr_info->data = vcpu->arch.osvw.status;
 		break;
+	case MSR_IA32_PERF_CAPABILITIES:
+		rdmsrl(MSR_IA32_PERF_CAPABILITIES, data);
+		break;
 	default:
 		if (kvm_pmu_is_valid_msr(vcpu, msr_info->index))
 			return kvm_pmu_get_msr(vcpu, msr_info->index, &msr_info->data);
-- 
2.7.4

